### PR TITLE
Fix NPE when reading an empty Iceberg table without snapshot ID

### PR DESCRIPTION
**Pull request by jackye1995**
Test plan - added test `testReadEmptyTable`, fixed related test failures

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by jackye1995** [comment]
@ChunxuTang @beinan

File: presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java [hunk 1]
```diff
@@ -152,18 +153,27 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
...
+    private Optional<Long> resolveSnapshotId(IcebergTableName name, Table table)
+    {
+        if (name.getSnapshotId().isPresent()) {
+            if (table.snapshot(name.getSnapshotId().get()) == null) {
+                throw new PrestoException(ICEBERG_INVALID_SNAPSHOT_ID, format("Invalid snapshot [%s] for table: %s", name.getSnapshotId().get(), table));
+            }
+            return name.getSnapshotId();
+        }
+        return Optional.ofNullable(table.currentSnapshot())
+                .map(Snapshot::snapshotId);
```
Comment (ChunxuTang) [inline presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java:174]: nit: Looks like these two lines are short enough to be put into one line. Your call.

**Review by ChunxuTang — COMMENTED** [review]
@jackye1995 LGTM! Thanks for your contribution!
BTW, would you mind updating the PR description following the guideline (https://github.com/prestodb/presto/blob/master/pull_request_template.md)? You may want to add a release note, or simply prefer no release note if the PR doesn't make significant changes.

**Review by beinan — APPROVED** [review]
lgtm

**Comment by beinan** [comment]
@jackye1995 could help update the description as @ChunxuTang said? thanks!

**Comment by jackye1995** [comment]
Thanks, I have updated the description section. Given this is not significant change I have removed the release note. @beinan @ChunxuTang

**Review by beinan — APPROVED** [review]
lgtm

**Pull request merged**